llarp/config/config.cpp

```diff
@@ -936,48 +933,95 @@ namespace llarp
         },
         [this](uint16_t arg) { public_port = net::port_t::from_host(arg); });
 
-    auto parse_addr_for_link = [net_ptr](
-                                   const std::string& arg, net::port_t default_port, bool inbound) {
-      std::optional<SockAddr> addr = std::nullopt;
+    // DISCUSS: does this need to be an optional?
+    auto parse_addr_for_link = [net_ptr](const std::string& arg) {
+      std::optional<oxen::quic::Address> maybe = std::nullopt;
+      std::string_view arg_v;
+
       // explicitly provided value
       if (not arg.empty())
       {
-        if (arg[0] == ':')
-        {
-          // port only case
-          default_port = net::port_t::from_string(arg.substr(1));
-          if (!inbound)
-            addr = net_ptr->WildcardWithPort(default_port);
-        }
-        else
-        {
-          addr = SockAddr{arg};
-          if (net_ptr->IsLoopbackAddress(addr->getIP()))
-            throw std::invalid_argument{fmt::format("{} is a loopback address", arg)};
-        }
+        arg_v = std::string_view{arg};
       }
-      if (not addr)
+
+      if (arg_v[0] == ':')
       {
-        // infer public address
-        if (auto maybe_ifname = net_ptr->GetBestNetIF())
-          addr = net_ptr->GetInterfaceAddr(*maybe_ifname);
+        uint16_t res = std::atoi(arg_v.substr(1).data());
```

Comment:
We have a nicer/safer/more modern `llarp::parse_int<uint16_t>` for integer parsing.  (There's another place for it below)